dbms/src/Processors/Formats/Impl/CSVRowInputFormat.cpp

```diff
@@ -269,6 +258,9 @@ bool CSVRowInputFormat::readRow(MutableColumns & columns, RowReadExtension & ext
 bool CSVRowInputFormat::parseRowAndPrintDiagnosticInfo(MutableColumns & columns, WriteBuffer & out)
 {
     const char delimiter = format_settings.csv.delimiter;
+    const String bad_delimiters = " \t\"'.UL";
+    if (bad_delimiters.find(delimiter) != String::npos)
+        out << "CSV format may not work correctly with delimiter '" << delimiter << "'\n";
```

Comment:
Can we just error out when the CSV row input format is created with these delimiters? It won't work anyway.

Reply:
With `.` as delimiter it will work if there are no Float or Decimal columns. With `U` and `L` it will work if `input_format_csv_unquoted_null_literal_as_null=0`. With `'` it will work if `format_csv_allow_single_quotes=0`